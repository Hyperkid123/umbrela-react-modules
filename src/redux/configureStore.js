import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
