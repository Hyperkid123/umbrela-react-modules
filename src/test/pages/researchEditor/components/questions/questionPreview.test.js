import React from 'react'
import QuestionPreview, {QuestionPreview as Snapshot} from '../../../../../pages/researchEditorModule/components/questions/questionPreview';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Question Preview component', () => {
  const initialState  = {
    questions: {
      activeQuestion: {
        questionId: 10
      }
    },
    options: {
      options: []
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<QuestionPreview store={store}/>);
    expect(wrapper).toBeDefined();
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'CloseQuestion',
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render order preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'OrderQuestion',
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render divide preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'DivideQuestion',
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render matrix preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'MatrixSingleQuestion',
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render customHelp', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'CloseQuestion',
          customHelp: 'foo help'
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render open question', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'OpenQuestion',
          customHelp: 'foo help'
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render image preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'MediaQuestion',
          url: 'foo.png'
        }}
        options={[]}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should set open answer', () => {
    const wrapper = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'OpenQuestion',
        }}
        options={[]}
        translate={jest.fn()}
      />
    );
    wrapper.instance().setOpenAnswer('foo');
    expect(wrapper.state().openAnswer).toEqual('foo');
  })
});
