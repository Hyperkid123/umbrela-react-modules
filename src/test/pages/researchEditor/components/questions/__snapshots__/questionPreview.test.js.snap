// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Preview component should render correctly empty 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <ClosePreview
    options={Array []}
    questionType="CloseQuestion"
  />
</div>
`;

exports[`Question Preview component should render customHelp 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <styled.div>
    <styled.div
      key="0"
    >
      foo help
    </styled.div>
  </styled.div>
  <ClosePreview
    options={Array []}
    questionType="CloseQuestion"
  />
</div>
`;

exports[`Question Preview component should render divide preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <DividePreview
    options={Array []}
  />
</div>
`;

exports[`Question Preview component should render image preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <LazyLoad
    offset={0}
    once={false}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <styled.img
      alt="foo.png"
      src="foo.png"
    />
  </LazyLoad>
  <ClosePreview
    options={Array []}
    questionType="MediaQuestion"
  />
</div>
`;

exports[`Question Preview component should render matrix preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <MatrixPreview
    options={Array []}
    questionType="MatrixSingleQuestion"
  />
</div>
`;

exports[`Question Preview component should render open question 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <styled.div>
    <styled.div
      key="0"
    >
      foo help
    </styled.div>
  </styled.div>
  <styled.div
    column={true}
    grow={true}
  >
    <TextField
      fullWidth={true}
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
    />
    <TextFieldComent
      alignRight={true}
      error={false}
      label="0 undefined 1500 undefined"
    />
  </styled.div>
</div>
`;

exports[`Question Preview component should render order preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.h2 />
  <OrderPreview
    options={Array []}
  />
</div>
`;
